pkg/accountmanager/poller_test.go

```diff
@@ -0,0 +1,358 @@
+			testSecretNamespacedName  = types.NamespacedName{Namespace: "namespace01", Name: "secret01"}
+			testCesNamespacedName     = types.NamespacedName{Namespace: "namespace01", Name: "Ces01"}
+			account                   *v1alpha1.CloudProviderAccount
+			accountPollerObj          *accountPoller
+			fakeClient                client.WithWatch
+			secret                    *corev1.Secret
+			ces                       *v1alpha1.CloudEntitySelector
+			mockCtrl                  *mock.Controller
+			mockCloudInterface        *cloud.MockCloudInterface
+			pollIntv                  uint
+		)
+		BeforeEach(func() {
+
+			var err error
+			newScheme := runtime.NewScheme()
+			utilruntime.Must(clientgoscheme.AddToScheme(newScheme))
+			utilruntime.Must(v1alpha1.AddToScheme(newScheme))
+
+			fakeClient = fake.NewClientBuilder().WithScheme(newScheme).Build()
+			mockCtrl = mock.NewController(GinkgoT())
+			mockCloudInterface = cloud.NewMockCloudInterface(mockCtrl)
+			cloudInventory := inventory.InitInventory()
+			accountPollerObj = &accountPoller{
+				log:       logf.Log,
+				Client:    fakeClient,
+				inventory: cloudInventory,
+				mutex:     sync.RWMutex{},
+			}
+
+			pollIntv = 1
+			account = &v1alpha1.CloudProviderAccount{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      testAccountNamespacedName.Name,
+					Namespace: testAccountNamespacedName.Namespace,
+				},
+				Spec: v1alpha1.CloudProviderAccountSpec{
+					PollIntervalInSeconds: &pollIntv,
+					AWSConfig: &v1alpha1.CloudProviderAccountAWSConfig{
+						Region: "us-east-1",
+						SecretRef: &v1alpha1.SecretReference{
+							Name:      testSecretNamespacedName.Name,
+							Namespace: testSecretNamespacedName.Namespace,
+							Key:       credentials,
+						},
+					},
+				},
+			}
+			credential := `{"accessKeyId": "keyId","accessKeySecret": "keySecret"}`
+			secret = &corev1.Secret{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      testSecretNamespacedName.Name,
+					Namespace: testSecretNamespacedName.Namespace,
+				},
+				Data: map[string][]byte{
+					"credentials": []byte(credential),
+				},
+			}
+			ces = &v1alpha1.CloudEntitySelector{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      testCesNamespacedName.Name,
+					Namespace: testCesNamespacedName.Namespace,
+					OwnerReferences: []v1.OwnerReference{
+						{
+							APIVersion: "crd.cloud.antrea.io/crdv1alpha1",
+							Kind:       "CloudProviderAccount",
+							Name:       testAccountNamespacedName.Name,
+						},
+					},
+				},
+				Spec: v1alpha1.CloudEntitySelectorSpec{
+					AccountName: testAccountNamespacedName.Name,
+					VMSelector: []v1alpha1.VirtualMachineSelector{
+						{
+							VpcMatch: &v1alpha1.EntityMatch{
+								MatchID: "xyz",
+							},
+							VMMatch: []v1alpha1.EntityMatch{
+								{
+									MatchID:   "xyz",
+									MatchName: "xyz",
+								},
+							},
+						},
+					},
+				},
+			}
+			Expect(err).ShouldNot(HaveOccurred())
```

Comment:
remove this as no op is being done here, also remove the declaration for error.

Reply:
Solved.